atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/atomic/CassImitatingConsensusForgettingStoreV4.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.atomic;
+
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+import com.palantir.atlasdb.keyvalue.api.KeyAlreadyExistsException;
+import com.palantir.common.streams.KeyedStream;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public final class CassImitatingConsensusForgettingStoreV4 extends CassandraImitatingConsensusForgettingStore {
+    static final byte[] IN_PROGRESS_MARKER = new byte[1];
+
+    public CassImitatingConsensusForgettingStoreV4(double probabilityOfFailure) {
+        super(probabilityOfFailure);
+    }
+
+    /**
+     * Atomically performs a read (potentially propagating newest read value) and if there is no value present on any of
+     * the nodes in a quorum, writes the value to those nodes. If there is a value present, throws a
+     * {@link KeyAlreadyExistsException} with detail.
```

Comment:
This probably needs to be updated given that the behaviour of the method has changed!